crates/fuel-core/src/query/balance/asset_query.rs

```diff
@@ -123,11 +138,18 @@ impl<'a> AssetsQuery<'a> {
                     } else {
                         return Err(anyhow::anyhow!("The coin is not a message").into());
                     };
-                    let message = self.database.message(&id)?;
+                    // TODO: Fetch message in a separate thread
```

Comment:
Since this is addressed in the next PR it's nice to have a reference here https://github.com/FuelLabs/fuel-core/pull/2340.

```suggestion
                    // TODO: Fetch message in a separate thread (https://github.com/FuelLabs/fuel-core/pull/2340)
```